main.py

```diff
@@ -1,52 +1,70 @@
 import uuid
 
 
-class Bank():
-    default_log='None'
-    def __init__(self,user_name,log=default_log):
-        self.user_name=user_name
-        self.log=str(log)
+class User():
+    def __init__(self,name):
+        self.name=name
         self.card=Card()
+
     def info(self):
-        print(f'User name:{self.user_name}')
-        return (self.card.info_card())
+        print(f'User name:{self.name}')
+        return (self.card.info())
+
+    def logg(self):
+        return(self.card.loggering())
 
 class Card:
     default_ballance=0
-    default_uuid=uuid.uuid4()
-    def __init__(self,uuid=default_uuid,ballance=default_ballance):
-        self.uuid=uuid
+    default_log=[]
+
+    def __init__(self,ballance=default_ballance,log=default_log):
```

Comment:
I'm not sure (should test it, but i did not do it yet) but please investigate - when you init different users with the same `default_log=[]` could it be (just assumption - i don't know) that you have just two different 'pointers' to the same peace of memory - and this could explain your problem with `# log function, main problem: logging at one log for two users at same time`
do you know for sure that log is shared?

Reply:
you are right, that is why problem exist, i fixed this.